lib/pipeline.js

```diff
@@ -499,6 +487,54 @@ class PipelineModel extends BaseModel {
         return admin;
     }
 
+    /**
+     * This function deletes admins who does not have proper
+     * GitHub permission, and returns an proper admin.
+     * @method getFirstAdmin
+     * @return {Promise}
+     */
+    async getFirstAdmin() {
+        /* eslint-disable no-restricted-syntax */
+        for (const username of Object.keys(this.admins)) {
+            try {
+                /* eslint-disable-next-line no-await-in-loop */
+                return await new Promise((resolve, reject) => {
+                    // Lazy load factory dependency to prevent circular dependency issues
+                    // https://nodejs.org/api/modules.html#modules_cycles
+                    /* eslint-disable global-require */
+                    const UserFactory = require('./userFactory');
+                    /* eslint-enable global-require */
+                    const factory = UserFactory.getInstance();
+
+                    const user = factory.get({
+                        username,
+                        scmContext: this.scmContext
+                    });
+
+                    user.then((realUser) => {
+                        realUser.getPermissions(this.scmUri)
+                            .then((permissions) => {
+                                if (!permissions.push) {
+                                    delete this.admins[username];
+
+                                    return reject();
+                                }
+
+                                return resolve(realUser);
+                            });
+                    });
+                });
+            } catch (e) {
+                /* eslint-disable no-console */
+                console.log(e);
```

Comment:
`undefined` may be printed if `return reject()` is called.
I think need more information for logging or delete this `console.log`.

Reply:
Thank you. I tried to delete, but `No catch clause` and `Empty block statement` is prohibited. Thus I add log message.

Reply:
I add `delete` non admin user in `catch`, and I could delete the `console.log`. I thought this is better.